File: App/src/main/java/com/fooddelivery/app/data/models/Payment.kt
package com.fooddelivery.app.data.models

import com.google.gson.annotations.SerializedName

data class Payment(
    @SerializedName("id")
    val id: Int,
    @SerializedName("order_id")
    val orderId: Int,
    @SerializedName("payment_number")
    val paymentNumber: String,
    @SerializedName("payment_method")
    val paymentMethod: String,
    @SerializedName("amount")
    val amount: Double,
    @SerializedName("status")
    val status: String,
    @SerializedName("transaction_id")
    val transactionId: String?,
    @SerializedName("payment_notes")
    val paymentNotes: String?,
    @SerializedName("paid_at")
    val paidAt: String?,
    @SerializedName("created_at")
    val createdAt: String,
    @SerializedName("updated_at")
    val updatedAt: String?
)

data class PaymentCreateRequest(
    @SerializedName("order_id")
    val orderId: Int,
    @SerializedName("payment_method")
    val paymentMethod: String,
    @SerializedName("transaction_id")
    val transactionId: String? = null,
    @SerializedName("payment_notes")
    val paymentNotes: String? = null
)

File: Backend-Api/schemas/payment.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime
from models.payment import PaymentMethod, PaymentStatus


class PaymentCreate(BaseModel):
    order_id: int
    payment_method: PaymentMethod
    transaction_id: Optional[str] = None
    payment_notes: Optional[str] = None


class PaymentResponse(BaseModel):
    id: int
    order_id: int
    payment_number: str
    payment_method: PaymentMethod
    amount: float
    status: PaymentStatus
    transaction_id: Optional[str]
    payment_notes: Optional[str]
    paid_at: Optional[datetime]
    created_at: datetime
    updated_at: Optional[datetime]

    class Config:
        from_attributes = True


class PaymentStatusUpdate(BaseModel):
    status: PaymentStatus
    transaction_id: Optional[str] = None

